assets/src/components/parts/janus-text-flow/Markup.tsx

```diff
@@ -64,7 +65,11 @@ const Markup: React.FC<any> = ({
   let processedText = text;
   // allow (authoring usually) skipping the template processing
   if (!displayRawText) {
-    processedText = templatizeText(text, state, env);
+    let updatedText = text;
+    expressions.forEach((exp: Record<string, string>) => {
+      updatedText = text.replaceAll(exp.expression, exp.newExpression);
+    });
+    processedText = templatizeText(updatedText, state, env);
```

Comment:
so I guess that I was thinking that `state` would just contain only the required values. i'm honestly having a hard time visualizing what is coming through for the expressions here that we are replacing in text. state usually comes from the snapshot being sent back from init, which has ALL of the state, are we still sending that? I was thinking that we'd just be able to send a list of required values and filter the snapshot being sent back. maybe you can describe what is happening here

Reply:
expressions is an array object that contains the original expression used in the text i.e. exp.expression    and exp.newExpression which is the expression with actual ownerId of the part component that is used in the variable.
 we are trying to replace the expression with the actual ownerId (i.e. the question Id) of the part component that is used in the variable.

Also, I think I can filter the state and send only the required variables.